src/main/kotlin/com/manimdsl/shapes/Shapes.kt

```diff
@@ -19,44 +20,47 @@ sealed class Shape {
     }
 }
 
-sealed class ShapeWithText : Shape() {
-    // Python shape classes which has shape and text members to manipulate color
+sealed class ShapeWithText : Shape()
 
-    fun getTextMObject(): String {
-        return "$ident.text"
-    }
-
-    fun getShapeMObject(): String {
-        return "$ident.shape"
-    }
+interface StyleableShape {
+    fun restyle(styleProperties: StylesheetProperty): List<String>
 }
 
 class Rectangle(
     override val ident: String,
     override val text: String,
     color: String? = null,
     textColor: String? = null,
-    textWeight: String? = null,
-    font: String? = null,
-) : ShapeWithText() {
+) : ShapeWithText(), StyleableShape {
     override val classPath: String = "python/rectangle.py"
     override val className: String = "Rectangle_block"
     override val pythonVariablePrefix: String = "rectangle"
 
     init {
         color?.let { style.addStyleAttribute(Color(it)) }
         textColor?.let { style.addStyleAttribute(TextColor(it)) }
-        textWeight?.let { style.addStyleAttribute(TextWeight(it)) }
-        font?.let { style.addStyleAttribute(Font(it)) }
     }
+
+    override fun restyle(styleProperties: StylesheetProperty): List<String> {
+        val instructions = mutableListOf<String>()
+
+        if (styleProperties.borderColor != null) {
+            instructions.add("FadeToColor($ident.shape, ${styleProperties.borderColor})")
+        }
+
+        if (styleProperties.textColor != null) {
+            instructions.add("FadeToColor($ident.text, ${styleProperties.textColor})")
+        }
+
+        return listOf("self.play(${instructions.joinToString(", ")})")
+    }
+
 }
 
 class CodeBlockShape(
     override val ident: String,
     lines: List<String>,
     textColor: String? = null,
-    textWeight: String? = null,
```

Comment:
What was the reason again for not allowing setting of text weight and font?

Reply:
We decided it was not necessary as it was changing fonts whenever it was moved around